encoding/codecv2/codecv2.go

```diff
@@ -410,3 +492,30 @@ func CalculatePaddedBlobSize(dataSize uint64) uint64 {
 func GetBlobDataProofArgs() (*abi.Arguments, error) {
 	return codecv1.GetBlobDataProofArgs()
 }
+
+// decompressScrollBatchBytes decompresses the given bytes into scroll batch bytes
+func decompressScrollBatchBytes(compressedBytes []byte) ([]byte, error) {
+	// decompress data in stream and in batches of bytes, because we don't know actual length of compressed data
+	var res []byte
+	readBatchSize := 131072
+	batchOfBytes := make([]byte, readBatchSize)
+
+	r := bytes.NewReader(compressedBytes)
+	zr, err := zstd.NewReader(r)
+	if err != nil {
+		return nil, err
+	}
+	defer zr.Close()
+
+	for {
+		i, err := zr.Read(batchOfBytes)
+		res = append(res, batchOfBytes[:i]...) // append already decoded bytes even if we meet error
+		if i < readBatchSize || err != nil {
```

Comment:
What about returning the error `if err != nil`? otherwise, the error is not handled.

Reply:
added a comment explaining this. the error is expected and means that decoder can't decode further compressed data. As explained at the beginning of method "decompress data in stream and in batches of bytes, **because we don't know actual length of compressed data**". So even after error we will have data written in `res` and should return this

Reply:
o i c. and yeah it makes sense to add a comment here.

Reply:
then what about early return before `res = append(res, batchOfBytes[:i]...)` when EOF?

Reply:
`i, err := zr.Read(batchOfBytes)` at this line we can get `err != nil` and `i > 0`. That means that decoder was able to decode `i` bytes and after that catch error. So we still need to append that decoded data and return only after that

Reply:
I added some comments explaining this all